langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AzureAiSearchEmbeddingStore.java

```diff
@@ -0,0 +1,426 @@
+                    String value = (String) innerAttribute.get("value");
+                    attributesMap.put(key, value);
+                }
+                Metadata langChainMetada = Metadata.from(attributesMap);
+                TextSegment embedded = TextSegment.textSegment(embbededContent, langChainMetada);
+                embeddingMatch = new EmbeddingMatch<>(score, embeddingId, embedding, embedded);
+            } else {
+                embeddingMatch = new EmbeddingMatch<>(score, embeddingId, embedding, null);
+            }
+            result.add(embeddingMatch);
+        }
+        return result;
+    }
+
+    private void addInternal(String id, Embedding embedding, TextSegment embedded) {
+        addAllInternal(
+                singletonList(id),
+                singletonList(embedding),
+                embedded == null ? null : singletonList(embedded));
+    }
+
+    private void addAllInternal(
+            List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        if (isNullOrEmpty(ids) || isNullOrEmpty(embeddings)) {
+            log.info("Empty embeddings - no ops");
+            return;
+        }
+        ensureTrue(ids.size() == embeddings.size(), "ids size is not equal to embeddings size");
+        ensureTrue(embedded == null || embeddings.size() == embedded.size(),
+                "embeddings size is not equal to embedded size");
+
+        List<Document> searchDocuments = new ArrayList<>();
+        for (int i = 0; i < ids.size(); ++i) {
+            Document document = new Document();
+            document.setId(ids.get(i));
+            document.setContent_vector(floatsArrayToList(embeddings.get(i).vector()));
+            if (embedded != null) {
+                document.setContent(embedded.get(i).text());
+                Document.Metadata metadata = new Document.Metadata();
+                metadata.setSource("langchain4j");
+                List<Document.Metadata.Attribute> attributes = new ArrayList<>();
+                for (Map.Entry<String, String> entry : embedded.get(i).metadata().asMap().entrySet()) {
+                    Document.Metadata.Attribute attribute = new Document.Metadata.Attribute();
+                    attribute.setKey(entry.getKey());
+                    attribute.setValue(entry.getValue());
+                    attributes.add(attribute);
+                }
+                metadata.setAttributes(attributes);
+                document.setMetadata(metadata);
+            } else {
+                document.setContent("");
+            }
+            searchDocuments.add(document);
+        }
+        List<IndexingResult> indexingResults = searchClient.uploadDocuments(searchDocuments).getResults();
+        for (IndexingResult indexingResult : indexingResults) {
+            if (!indexingResult.isSucceeded()) {
+                log.error("Failed to add embedding: {}", indexingResult.getErrorMessage());
```

Comment:
should we maybe throw an exception here instead of logging?

Reply:
yes